MapboxNavigation/NavigationMapView.swift

```diff
@@ -442,39 +469,139 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
-            let lineCasing = navigationMapViewDelegate?.navigationMapView(self, routeCasingStyleLayerWithIdentifier: StyleLayerIdentifier.routeCasing, source: lineCasingSource) ?? routeCasingStyleLayer(identifier: StyleLayerIdentifier.routeCasing, source: lineSource)
-            
+            // Otherwise, create them for the first time.
+
+            // Source layer for main route + alternative routes
+            let allRoutesSource = MGLShapeSource(identifier: SourceIdentifier.allRoutes, shape: polylines, options: [.lineDistanceMetrics: true])
+            style.addSource(allRoutesSource)
+
+            generateTrafficGradientStops(for: mainRoute)
+
+            let mainRouteLayer = navigationMapViewDelegate?.navigationMapView(self, mainRouteStyleLayerWithIdentifier: StyleLayerIdentifier.mainRoute, source: allRoutesSource) ?? mainRouteStyleLayer(identifier: StyleLayerIdentifier.mainRoute, source: allRoutesSource)
+            let mainRouteCasingLayer = navigationMapViewDelegate?.navigationMapView(self, mainRouteCasingStyleLayerWithIdentifier: StyleLayerIdentifier.mainRouteCasing, source: allRoutesSource) ?? mainRouteCasingStyleLayer(identifier: StyleLayerIdentifier.mainRouteCasing, source: allRoutesSource)
+            let alternateRoutesLayer = navigationMapViewDelegate?.navigationMapView(self, alternateRouteStyleLayerWithIdentifier: StyleLayerIdentifier.alternateRoutes, source: allRoutesSource) ?? alternateRouteStyleLayer(identifier: StyleLayerIdentifier.alternateRoutes, source: allRoutesSource)
+             let alternateRoutesCasingLayer = navigationMapViewDelegate?.navigationMapView(self, mainRouteCasingStyleLayerWithIdentifier: StyleLayerIdentifier.alternateRoutesCasing, source: allRoutesSource) ?? alternateRouteCasingStyleLayer(identifier: StyleLayerIdentifier.alternateRoutesCasing, source: allRoutesSource)
+
+            // Add all the layers in the correct order
             for layer in style.layers.reversed() {
                 if !(layer is MGLSymbolStyleLayer) &&
                 layer.identifier != StyleLayerIdentifier.arrow && layer.identifier != StyleLayerIdentifier.arrowSymbol && layer.identifier != StyleLayerIdentifier.arrowCasingSymbol && layer.identifier != StyleLayerIdentifier.arrowStroke && layer.identifier != StyleLayerIdentifier.waypointCircle {
-                    style.insertLayer(line, below: layer)
-                    style.insertLayer(lineCasing, below: line)
+                    style.insertLayer(mainRouteLayer, below: layer)
+                    style.insertLayer(mainRouteCasingLayer, below: mainRouteLayer)
+                    style.insertLayer(alternateRoutesLayer, below: mainRouteCasingLayer)
+                    style.insertLayer(alternateRoutesCasingLayer, below: alternateRoutesLayer)
```

Comment:
This is the part that fixes #1434. I’m a little uneasy about the fact that we’ll be inserting four layers for the route line, but it looks like it’ll be straightforward (and maybe not critical) to recombine a couple of them by conditionalizing properties (or individual stop values) on whether an `isAlternativeRoute` or `isCasing` attribute is set. Not necessary for landing this PR though.